pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -629,3 +630,147 @@ var (
+	// deletion of ThisWorkspace resources
+	ThisWorkspaceFinalizer = "tenancy.kcp.dev/thisworkspace"
+)
+
+// ThisWorkspace describes the current workspace.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Type",type=string,JSONPath=`.spec.type.name`,description="Type of the workspace"
+// +kubebuilder:printcolumn:name="Phase",type=string,JSONPath=`.status.phase`,description="The current phase (e.g. Scheduling, Initializing, Ready)"
+// +kubebuilder:printcolumn:name="URL",type=string,JSONPath=`.status.URL`,description="URL to access the workspace"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+type ThisWorkspace struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +optional
+	Spec ThisWorkspaceSpec `json:"spec, omitempty"`
+	// +optional
+	Status ThisWorkspaceStatus `json:"status,omitempty"`
+}
+
+// ThisWorkspaceSpec is the specification of the ThisWorkspace resource.
+type ThisWorkspaceSpec struct {
+	// type defines properties of the workspace both on creation (e.g. initial
+	// resources and initially installed APIs) and during runtime (e.g. permissions).
+	// If no type is provided, the default type for the workspace in which this workspace
+	// is nesting will be used.
+	//
+	// The type is a reference to a ClusterWorkspaceType in the listed workspace, but
+	// lower-cased. The ClusterWorkspaceType existence is validated at admission during
+	// creation. The type is immutable after creation. The use of a type is gated via
+	// the RBAC clusterworkspacetypes/use resource permission.
+	//
+	// +optional
+	Type ClusterWorkspaceTypeReference `json:"type,omitempty"`
+
+	// owner is a reference to a resource controlling the life-cycle of this workspace.
+	// On deletion of the ThisWorkspace, the finalizer tenancy.kcp.dev/thisworkspace is
+	// removed from the owner.
+	//
+	// When this object is deleted, but the owner is not deleted, the owner is deleted
+	// too.
+	//
+	// +optional
+	Owner *ThisWorkspaceOwner `json:"owner,omitempty"`
+}
+
+// ThisWorkspaceOwner is a reference to a resource controlling the life-cycle of a ThisWorkspace.
+type ThisWorkspaceOwner struct {
+	// apiVersion is the group and API version of the owner.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([^/]+/)?[^/]+$`
+	APIVersion string `json:"apiVersion"`
+
+	// resource is API resource to access the owner.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Resource string `json:"resource"`
+
+	// name is the name of the owner.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
```

Comment:
If we care, we can do a lot stricter validation for name/namespace/resource/apiVersion